TrackingTools/TrackRefitter/src/TrackTransformerForCosmicMuons.cc

```diff
@@ -341,7 +341,7 @@ vector<Trajectory> TrackTransformerForCosmicMuons::transform(const reco::Track&
     return vector<Trajectory>();
   }
   
-  return trajectoriesSM;
+  return std::move(trajectoriesSM);
```

Comment:
I recall that in similar other cases we switched to lvalue instead of introducing a move.

Reply:
The warning is
```
TrackingTools/TrackRefitter/src/TrackTransformerForCosmicMuons.cc:344:10: warning: local variable 'trajectoriesSM' will be copied despite being returned by name [-Wreturn-std-move]
```
I suppose this is because of it being an rvalue-reference `vector<Trajectory> && trajectoriesSM = ...` on line 337 above.

It would likely be clearer to drop the rvalue reference, i.e. change to `vector<Trajectory> trajectoriesSM = ...`, and keep returning the `trajectoriesSM` by value (relying on NRVO).

Reply:
@makortel 
thanks.
that's exactly what I meant, in a telegram style.

Reply:
@smuzaffar 
please clarify if you are going to update this PR following the review comments.
Thank you.

Reply:
@slava77 , @makortel I have updated it to use `const vector<Trajectory> &trajectoriesSM`. Let me know if you really want to use `vector<Trajectory> trajectoriesSM` instead.

By thr way, you we need line number 341? why not just return trajectoriesSM even if it is empty?

Reply:
I believe it really should be `vector<Trajectory>` (i.e. value instead of const reference) for the NRVO to apply (or at least that's what a quick test on godbolt suggests). 

> By thr way, you we need line number 341? why not just return trajectoriesSM even if it is empty?

I was wondering the same and would just return `trajectoriesSM` (probably some historical leftover).